handler/core/strategy/hmacsha.go

```diff
@@ -0,0 +1,36 @@
+package strategy
+
+import (
+	"github.com/ory-am/fosite"
+	"github.com/ory-am/fosite/enigma"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+type HMACSHAStrategy struct {
+	Enigma enigma.HMACSHAEnigma
+}
+
+func (h HMACSHAStrategy) GenerateAccessToken(_ context.Context, _ *http.Request, requester fosite.AccessRequester, _ interface{}) (token string, signature string, err error) {
+	return h.Enigma.Generate(requester.GetClient().GetHashedSecret())
```

Comment:
My intuition says "keep it". Since there is always a good idea for future
development to have one backup of passing data along the call-chain ;)
Nothing that ever should be a praxis, but one that just might save that
situations from alternating any public interfaces.


Mvh

*Alexander Widerberg*
*Vakans Linköpings universitet i programvaruutveckling*


*DatateknologsektionenLinköpings UniversitetMobil: 0763-471640*

2016-01-14 21:50 GMT+01:00 Aeneas <notifications@github.com>:

> In handler/core/strategy/hmacsha.go
> <https://github.com/ory-am/fosite/pull/20#discussion_r49782997>:
>
> > @@ -0,0 +1,36 @@
> > +package strategy
> > +
> > +import (
> > +	"github.com/ory-am/fosite"
> > +	"github.com/ory-am/fosite/enigma"
> > +	"golang.org/x/net/context"
> > +	"net/http"
> > +)
> > +
> > +type HMACSHAStrategy struct {
> > +	Enigma enigma.HMACSHAEnigma
> > +}
> > +
> > +func (h HMACSHAStrategy) GenerateAccessToken(_ context.Context, _ *http.Request, requester fosite.AccessRequester, _ interface{}) (token string, signature string, err error) {
> > +	return h.Enigma.Generate(requester.GetClient().GetHashedSecret())
>
> by the way i think it would be okay to remove the context variable as I'm
> not sure that the generators/ validators really need it. however, I do not
> want to change the interface after this PR here any more so I can't really
> decide if keep or kick :D
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ory-am/fosite/pull/20/files#r49782997>.
>